library/std/src/sys/unix/fs.rs

```diff
@@ -1231,6 +1252,22 @@ impl File {
                 })?;
                 Ok(())
             } else {
+                #[cfg(all(target_os = "linux", target_env = "gnu", target_pointer_width = "32", not(target_arch = "riscv32")))]
```

Comment:
These changes are to 32-bit RISC-V Linux and seem to be unrelated to this PR?

Reply:
Yes, this should be split into a separate PR since it doesn't involve any API changes.